sdk/lib/drivers/sound/mmebuddy/mmewrap.c

```diff
@@ -365,3 +365,85 @@ MmeGetPosition(
     return Result;
 }
 
+
+MMRESULT
+MmeGetVolume(
+    _In_ MMDEVICE_TYPE DeviceType,
+    _In_ DWORD DeviceId,
+    _In_ DWORD_PTR PrivateHandle,
+    _Out_ PDWORD_PTR pdwVolume)
+{
+    MMRESULT Result;
+    PSOUND_DEVICE_INSTANCE SoundDeviceInstance;
+    PSOUND_DEVICE SoundDevice;
+    PMMFUNCTION_TABLE FunctionTable;
+
+    /* Sanity check */
+    SND_ASSERT(DeviceType == AUX_DEVICE_TYPE ||
+               DeviceType == MIDI_OUT_DEVICE_TYPE ||
+               DeviceType == WAVE_OUT_DEVICE_TYPE);
+
+    VALIDATE_MMSYS_PARAMETER(PrivateHandle);
+    SoundDeviceInstance = (PSOUND_DEVICE_INSTANCE)PrivateHandle;
+
+    if (!IsValidSoundDeviceInstance(SoundDeviceInstance))
+        return MMSYSERR_INVALHANDLE;
+
+    Result = GetSoundDeviceFromInstance(SoundDeviceInstance, &SoundDevice);
+    if (!MMSUCCESS(Result))
+        return TranslateInternalMmResult(Result);
+
+    Result = GetSoundDeviceFunctionTable(SoundDevice, &FunctionTable);
+    if (!MMSUCCESS(Result))
+        return TranslateInternalMmResult(Result);
+
+    if (!FunctionTable->GetVolume)
+        return MMSYSERR_NOTSUPPORTED;
+
+    /* Call the driver */
+    Result = FunctionTable->GetVolume(SoundDeviceInstance, DeviceId, (PDWORD)pdwVolume);
+
+    return Result;
+}
+
+
```

Comment:
```suggestion
```
yes, I admit: I am nitpicking here. But just because I do know that sooner or later I will want to port your PR back ;-)